sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockingEventProcessorClientTest.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Azure.Messaging.EventHubs.Processor.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample07_MockingEventProcessorClientTest
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task BasicEventProcessing()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_MockingBasicEventProcessing
+
+            var mockEventProcessorClient = new Mock<EventProcessorClient>();
+            mockEventProcessorClient
+                .Setup(c => c.StartProcessingAsync(It.IsAny<CancellationToken>()))
+                .Returns(Task.CompletedTask);
+            mockEventProcessorClient
+                .Setup(c => c.StopProcessingAsync(It.IsAny<CancellationToken>()))
+                .Returns(Task.CompletedTask);
+
```

Comment:
@jsquire ,

> How do I mock invoking my event and error handlers for the EventProcessorClient? (separate sample in the processor package)

Demonstrate using the EventHubsModelFactory to create the types needed by the event args

I'm having trouble understanding the scenario here, could you please help me out, what is it that we should be mocking?
processEventHanlder and processErrorHandler?


>